src/clj/puppetlabs/trapperkeeper/services/metrics/jolokia.clj

```diff
@@ -73,11 +75,29 @@
 
 (defn create-servlet
   "Builds a Jolokia Servlet that uses Clojure logging."
-  []
-  (proxy [AgentServlet] []
-    ;; NOTE: An alternative to this method override would be to use defrecord
-    ;; to create a class that can be set as `:logHandlerClass` in the servlet
-    ;; configuration. This requires AOT compilation for the namespace defining
-    ;; the record so that Jolokia can find the resulting class.
-    (createLogHandler [_ _]
-      (create-logger))))
+  [auth-check-fn]
+  (let [check-auth (if auth-check-fn
+                     (fn [^HttpServletRequest request]
+                      (auth-check-fn
+                       (ring-servlet/build-request-map request)))
+                     (fn [^HttpServletRequest request]
+                       {:authorized true :message ""}))]
+    (if auth-check-fn
+      (log/info "Metrics access control using trapperkeeper-authorization is enabled.")
+      (log/warn "Metrics access control using trapperkeeper-authorization is disabled. Add the authorization service to the trapperkeeper bootstrap configuration file to enable it."))
```

Comment:
Should this message reference the exact name/location of the trapperkeeper bootstrap config?

Reply:
No, it varies depending on service (and could actually be multiple files).

Reply:
ahh interesting.. I'm fine with this if we think people will really know what this is talking about, I just know I get frustrating when now I need to google something just to interpret the error message, ya know?

Reply:
Yeah, before I responded with that abrupt answer I searched to see if there was a page we could link folks to that would help them figure it out and I couldn't find anywhere that we publicly document our bootstrap config file format or expected locations.

Also, all the tk-auth info I could find is under Puppet Server. I don't know of any more useful info we could point folks to right now (but I don't think we know where the bootstrap config file they should edit is from within this service)...

Reply:
Are they at least all called `bootstrap.cfg` or something? Or would they also have different base names?

Reply:
It looks like in Puppet Server FOSS we use the "system" vs "user" configs within a services.d/ s (see both of these dirs [here](https://github.com/puppetlabs/puppetserver/tree/master/ezbake)).

It looks like PE uses "bootstrap.cfg" rooted in each services higher level config dir. ([here](https://github.com/puppetlabs/puppet-enterprise-modules/blob/master/modules/puppet_enterprise/manifests/trapperkeeper/bootstrap_cfg.pp))

Who is this message directed at? End users or developers/consumers of the library? I think the current description is good enough for developers. For end users FOSS Server users will just get this change automatically since we already use tk-auth. PE users should have one or more toggles that I think Patrick is working on in another PR. I think FOSS PDB are the folks who will most likely see this and I don't know what we'd want to put there for them, nor am I sure that we should PDB specific info in an upstream library.

Reply:
I agree that we shouldn't put any project-specific info in here.

So would it be accurate to say "Add the authorization service to the trapperkeeper `bootstrap.cfg` file" ? I see that there are other `.cfg` files in puppetserver, but if we'll be handling that part anyway.. ¯\_(ツ)_/¯ 

Sorry, didn't mean to turn this into a long tangent, just want to be as clear as possible for users (and future me 😜).